### ISSUE TITLE
0.2.0 incompatible with rollup

### ISSUE THREAD

**Issue by ghempton** [systemjs/plugin-json#11]
When building a static build with rollup optimizations, rollup is unable to detect named exports from .json files. This works fine if the json plugin is downgraded to 0.1.0.

**Comment by guybedford** [comment]
The code at https://github.com/systemjs/plugin-json/blob/master/json.js#L9 would need to be adjusted to deal with generating a named export variable per JSON property.

Basically we want a JSON file like:

``` json
{
  "some": "export",
  "another": {
    "export": "object"
  }
}
```

To be output as ES:

``` javascript
export let some = "export";
export let another = {
  "export": "object"
};
export default {
  some: some,
  another: another
};
```

Then Rollup would be able to handle things correctly.

This serialization is currently done naively in https://github.com/systemjs/plugin-json/blob/master/json.js#L9 with the PR by @sergei-startsev.

If either of you is interested in working on the serialization to the above form a new PR would be great.

**Comment by ghempton** [comment]
I'm not even concerned about having it be properly rolled up, just not having it explode. How does the 0.1 version of the plugin get around this?

**Comment by guybedford** [comment]
The 0.1 version doesn't support Rollup at all.

**Comment by ghempton** [comment]
With 0.1 I can do `buildStatic({rollup: true})` and there are no errors (and it appears that other parts of my code are rolled up). With 0.2, the same call fails with the rollup issue. It seems like 0.2 should default to a different module format so that rollup can still succeed with other modules?

**Comment by guybedford** [comment]
Added this support in https://github.com/systemjs/plugin-json/commit/75d8a6d0ee5efd06d982001f770524985c4b8cac. Let me know if this works for you. It's actually quite nice to be able to have Rollup support for partial JSON includes.

**Comment by Hypercubed** [comment]
Just a quick note before I forget... this breaks JSON files that are top level arrays.

**Comment by Silviu-Marian** [comment]
![image](https://cloud.githubusercontent.com/assets/1747499/21079947/ad99715e-bfaa-11e6-8803-507ea83b7a1b.png)
And also this

**Comment by Silviu-Marian** [comment]
Actually turns out it's just a bad RegExp in there:

`return exportName.match(/[a-zA-Z_$][0-9a-zA-Z_$]*/);`

should probably be

`return exportName.match(/^[a-zA-Z_$][0-9a-zA-Z_$]*$/);`

to match the whole thing not just parts of it because right now it returns true for everything

**Comment by guybedford** [comment]
@Silviu-Marian thanks I've updated the regex and released a 0.2.3.
